### PR TITLE
Remove `proxy` service and make API NGINX setup similar to the frontend

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Partially addresses #4108 by @dhruvkb

## Description

This PR
- removes the service `proxy` which was meant to test the API running under HTTPS locally. This was rarely or maybe even never used.
  - also removes the documentation and scaffolding files related to this service
- makes the API NGINX configuration identical to the frontend
  - also uses the `snippets/` pattern used in the frontend for common config in API locations
- updates Docker Compose profiles and `just` recipes so that the `api_extra` profile depends on the `api` profile

## Testing Instructions

Check out this PR and verify that the `nginx` service builds and runs as expected.

```bash
just build nginx && just api/up-extra && just logs nginx
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`./ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`./ov just catalog/generate-docs media-props`
      for the catalog or `./ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/justfile [hunk 1]
```diff
@@ -50,6 +50,7 @@ up *flags:
 
 # Bring up services specific to the API profile, in addition to the API server
 up-extra *flags:
+    just up {{ flags }}
```
Comment (AetherUnbound) [inline api/justfile:53]: If this is also calling `just ../up` with the correct compose profile, why should this also be called? If we needed it, shouldn't we just add `api` to the list of `COMPOSE_PROFILES` to spin up?

File: api/nginx/snippets/django_location.conf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```
Comment (AetherUnbound) [inline api/nginx/snippets/django_location.conf:1]: This is great!

**Review by AetherUnbound — COMMENTED** [review]
The code changes look good to me and everything (but the removed container) still runs fine for me. My concern is though that, even if we don't use that service often, we still added it for a reason. I seem to remember it being related to an incident or some issue. Can we make sure we still have no use for the HTTPS setup before removing it?

**Comment by dhruvkb** [comment]
@AetherUnbound yes, we had an issue where the links to the thumbnail and detail endpoints were `http://` instead of `https://`. But in the end, we fixed that using a custom field that automatically rewrites the URL scheme based on a value from the settings.

https://github.com/WordPress/openverse/blob/90f87b8712b50cf05401de91eb5805b067efeb85/api/api/serializers/fields.py#L9

Since that issue was resolved, I have not once run the API locally as HTTPS and do not anticipate any future situation where we would want or need to.

Additionally removing this removes a dependency on `mkcert` and brings us one step closer to `ov` having everything we need to run Openverse locally.

File: api/justfile (hunk 1 above)
Comment (dhruvkb) [reply api/justfile]: You are right! I had not realised that. Fixed in 8379b8b.

Comment (dhruvkb) [reply api/justfile]: You are right! I had not realised that. Fixed in 8379b8b.

**Comment by AetherUnbound** [comment]
> @AetherUnbound yes, we had an issue where the links to the thumbnail and detail endpoints were `http://` instead of `https://`. But in the end, we fixed that using a custom field that automatically rewrites the URL scheme based on a value from the settings.

Got it, thank you for giving me context on that! This definitely gives me more confidence in being able to remove this service 😄

**Review by AetherUnbound — APPROVED** [review]

**Pull request merged**